lua/entities/ent_jack_sleepingbag.lua

```diff
@@ -0,0 +1,86 @@
+ENT.Type 			= "anim"
+ENT.PrintName		= "Sleeping Bag"
+ENT.Author			= "Basipek"
+ENT.Category			= "JMod - Misc"
+ENT.Spawnable			= true
+ENT.AdminSpawnable		= true
+
+
+if (CLIENT) then
+	function ENT:Draw()
+		self.Entity:DrawModel()
+	end
+elseif (SERVER) then
+
+	function ENT:Initialize()
+		self.Entity:SetModel("models/props_equipment/sleeping_bag1.mdl")
+		self.Entity:PhysicsInit(SOLID_VPHYSICS)
+		self.Entity:SetMoveType(MOVETYPE_VPHYSICS)
+		self:SetSolid( SOLID_VPHYSICS )
+		self.jackSleepingBagOwner=nil
+		local phys=self.Entity:GetPhysicsObject()
+		if phys:IsValid()then
+			phys:Wake()
+			phys:SetMass(35)
+			self:SetColor(Color(100,100,100))
+		end
+		
+		self:SetUseType(SIMPLE_USE)
+		
+		self.nextSpawnTime=0
+	end
+
+	function ENT:Use(a)
+		local Alt = a:KeyDown(JMod.Config.General.AltFunctionKey)
+		if (a:IsPlayer() and Alt) then
+			--Turn into rolled bedroll
+		elseif a:IsPlayer() then 
+			if (self.jackSleepingBagOwner and IsValid(self.jackSleepingBagOwner)) then
+				if (a~=self.jackSleepingBagOwner) then
+					a:PrintMessage(HUD_PRINTCENTER,"This bed is already claimed!")
+				else
+					a:PrintMessage(HUD_PRINTCENTER,"This bed is already yours!")
+				end
+			else
+				if (IsValid(a.jackSleepingBag)) then a.jackSleepingBag.jackSleepingBagOwner=nil;a.jackSleepingBag:SetColor(Color(100,100,100)) end
+				a:PrintMessage(HUD_PRINTCENTER,"Bed has been claimed!")
+				self.jackSleepingBagOwner=a
+				a.jackSleepingBag=self
```

Comment:
change `.jackSleepingBag` to `.JModSpawnPointEntity` to properly distinguish it as part of of this addon